src/lib/storage/segment_access_counter.hpp

```diff
@@ -0,0 +1,98 @@
+#pragma once
+
+#include <atomic>
+#include <map>
+#include <memory>
+#include <string>
+
+#include "storage/pos_list.hpp"
+#include "types.hpp"
+
+namespace opossum {
+class Table;
+
+class SegmentAccessCounter {
```

Comment:
Could you add some documentation in this file?

Reply:
What does "how" mean? Maybe you could also add who increments the counter and how it is used (i.e., consumers have to track differences).